### PR TITLE
First implementation of influx sink.

### PR DESCRIPTION
**Pull request by 1p4pk**

### CODE REVIEW

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBCommittableSerializer.java [hunk 1]
```diff
@@ -33,14 +33,13 @@ public int getVersion() {
...
+    public byte[] serialize(final Void obj) throws IOException {
+        return SimpleVersionedStringSerializer.INSTANCE.serialize(String.valueOf(obj));
     }
 
     @Override
-    public String deserialize(final int version, final byte[] serialized) throws IOException {
-        // Here we should deserialize to InfluxDB data point
-        return SimpleVersionedStringSerializer.INSTANCE.deserialize(version, serialized);
+    public Void deserialize(final int version, final byte[] serialized) throws IOException {
+        // return SimpleVersionedStringSerializer.INSTANCE.deserialize(version, serialized);
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBCommittableSerializer.java]:
```suggestion
```

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/writer/InfluxDBSchemaSerializer.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.streaming.connectors.influxdb.sink.writer;
+
+import com.influxdb.client.write.Point;
+import java.io.Serializable;
+
+public interface InfluxDBSchemaSerializer<IN> extends Serializable {
+    /**
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/writer/InfluxDBSchemaSerializer.java:26]:
```suggestion

    /**
```

File: flink-next-gen-connector-influxdb/src/test/java/org/apache/flink/streaming/connectors/InfluxDBSinkITCase.java [hunk 3]
```diff
@@ -20,70 +20,70 @@
...
+     *     1L,2L,3L           "Test,LongValue=1 fieldKey="fieldValue"",
+     *                        "Test,LongValue=2 fieldKey="fieldValue"",
+     *                        "Test,LongValue=3 fieldKey="fieldValue"",
      *     (source1/1) -----> (sink1/1)
      * </pre>
      */
     @Test
     void testIncrementPipeline() throws Exception {
-        final StreamExecutionEnvironment env = this.buildStreamEnv();
+        final StreamExecutionEnvironment env = buildStreamEnv();
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/test/java/org/apache/flink/streaming/connectors/InfluxDBSinkITCase.java:71]:
```suggestion
        final StreamExecutionEnvironment env = this.buildStreamEnv();
```

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBSink.java [hunk 4]
```diff
@@ -22,49 +22,72 @@
...
+
+    private InfluxDBSink(
+            final InfluxDBSchemaSerializer<IN> influxDBSchemaSerializer,
+            final InfluxDBConfig influxDBConfig,
+            @Nullable final SimpleVersionedSerializer<IN> writerStateSerializer,
+            @Nullable final Committer<Void> committer,
+            @Nullable final SimpleVersionedSerializer<Void> committableSerializer,
+            @Nullable final GlobalCommitter<Void, Void> globalCommitter,
+            @Nullable final SimpleVersionedSerializer<Void> globalCommittableSerializer) {
+        this.inInfluxDBSchemaSerializer = influxDBSchemaSerializer;
```
Comment (Shark) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBSink.java]: Refactor `influxDBSchemaSerializer`

File: flink-next-gen-connector-influxdb/src/test/java/org/apache/flink/streaming/connectors/InfluxDBSinkITCase.java [hunk 5]
```diff
@@ -92,15 +92,38 @@ void testIncrementPipeline() throws Exception {
...
-    private StreamExecutionEnvironment buildStreamEnv() {
+    private static StreamExecutionEnvironment buildStreamEnv() {
         final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         env.setRuntimeMode(RuntimeExecutionMode.STREAMING);
         env.setParallelism(1);
         env.enableCheckpointing(100);
         return env;
     }
+
+    private static Queue<String> queryWrittenData(final InfluxDBConfig influxDBConfig) {
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/test/java/org/apache/flink/streaming/connectors/InfluxDBSinkITCase.java]: Maybe use a simple List<String> interface here. TODO for your next branch not here anymore.

**Review by Shark — APPROVED** [review]
💖

**Pull request merged**